### ISSUE TITLE
Neural network outputs are the same as the bias.

### ISSUE THREAD

**Issue by BenjaminZ** [deeplearning4j/deeplearning4j#827]
My output of the network are the same as the bias numbers, regards of the inputs.
I use my own DataFetcher to feed the data.
Here is the repository:
https://github.com/BenjaminZ/nntest

**Comment by AlexDBlack** [comment]
so thanks for the repository, it made it a lot easier to debug

so turns out the culprit here was dropconnect (plus poor explanation/javadoc on our part)

so .useDropConnect(true) uses the dropout coefficient
- to use dropconnect: .useDropConnect(true).dropOut(probabilityOfRetainingConnection)
- to use dropout: .dropOut(probabilityOfRetainingActivation)

now, dropout coefficient defaults to 0.0, which means your use of useDropConnect(true) without dropOut(0.5) (or whatever value) is equivalent to saying "use drop connect with 0 probability of retaining a weight during forward pass" (i.e., all weights are temporarily set to 0 during training). Funnily enough, that doesn't work so well.

the solution obviously is to set .dropOut(0.5) for example (on the layer itself), or just not use drop connect

I'll update the javadoc and see what we can do in the way of validity checks to avoid this sort of thing in the future
